.circleci/config.yml

```diff
@@ -7,6 +7,24 @@ executors:
   cif_executor:
     docker:
       - image: circleci/openjdk:11-jdk-node
+    environment:
+      # either snapshot or production
+      DEPENDENCIES: snapshot
```

Comment:
That's the only part I don't really like, because it's something we have to change manually and it means that the `main` branch might sometimes not build OOTB with released dependencies. I'm not a big fan of keeping a `dev` branch, but maybe in this case it would be best. We'd open PRs against `main` or against `dev` (depending on the need for the snapshot dependency or not), and set that `snapshot` mode automatically only for the `dev` branch. WDYT?

Reply:
I imagine the workflow like this:
* Develop feature for CIF core components and make sure it was merged to `master`.
* Create feature branch in Venia, add snapshot dependency and change build to `snapshot`. Now the PR can be merged and CircleCI still passes.
* Venia `master` builds, but releases are blocked.
* Release CIF core components. Update versions in Venia project and update CircleCI back to `release`.
* Release Venia.

Reply:
>  add snapshot dependency and change build to snapshot

I wonder if we can automate this via some GitHub API. Detect that the current folder is part of a git repo and what branch we're on.

Reply:
CircleCI environment variables give you details on the branch / PR the job is on.

Reply:
Actually we'd also have to set the `-SNAPSHOT` version at https://github.com/adobe/aem-cif-guides-venia/blob/main/pom.xml#L52 to make sure we have the snapshots for the components.

Also while I understand your proposed workflow, I'd propose that we use a `dev` branch where we automatically set the `DEPENDENCIES: snapshot` variable (hopefully this is possible in CircleCI) and also the `-SNAPSHOT` version for the CIF components dependency in the main `pom.xml`. Of course we'd have to find a way to do that, maybe we can do that with a dedicated build profile only set on `dev`, something like `mvn clean install -Pcloud,snapshot`.

Reply:
> I prefer this so we don't have to bother about "switching" the toggle before a release: although we'd have to be careful not to merge this into master, so in the end it's maybe not really better ...
I managed to confuse myself ;-) if we automate the "toggles" by detecting that we are on a `dev` branch, then we don't to be careful when merging into `master`.